main.tf

```
@@ -16,7 +16,7 @@ data "aws_iam_policy_document" "default" {
       "s3:PutObject"
     ]
     resources = [
-      "arn:${data.aws_partition.current.partition}:s3:::${module.this.id}/*",
+      "arn:${data.aws_partition.current.partition}:s3:::${var.bucket_name}/*",
```

If the `var.bucket_name` isn't used, then `module.this.id` should be used instead

Add the following at the top (it's a [copy](https://github.com/cloudposse/terraform-aws-s3-log-storage/blob/main/main.tf#L1-L15) from the dependent module).

```hcl
locals {
  # We do not use coalesce() here because it is OK if local.bucket_name is empty.
  bucket_name = var.bucket_name == null || var.bucket_name == "" ? module.bucket_name.id : var.bucket_name
}

module "bucket_name" {
  source  = "cloudposse/label/null"
  version = "0.25.0"

  enabled = local.enabled && try(length(var.bucket_name) == 0, false)

  id_length_limit = 63 # https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html

  context = module.this.context
}
```

```suggestion
      "arn:${data.aws_partition.current.partition}:s3:::${local.bucket_name}/*",
```

---

Thank you @nitrocode for the completing suggestions.
I just had to add `enabled = module.this.enabled` to `locals` as an addition.